src/Http/ControllerFactory.php

```diff
@@ -33,14 +33,40 @@ class ControllerFactory
      */
     public function create(ServerRequest $request, Response $response)
     {
-        $pluginPath = $controller = null;
+        $controller = null;
+        if ($request->getParam('controller')) {
+            $controller = $request->getParam('controller');
+        }
```

Comment:
Trying to keep it simple as $controller is used in below https://github.com/cakephp/cakephp/pull/10558/files/4a49f1e8393bb259cf36c4455d1b42f87bdc4e63#diff-4c9ecd90d3740d54e09028ff7be12e1eR49

But makes sense to allow others to use the new getControllerClass method without passing the controller.